### PR TITLE
SUP-97 Changed resource upload file name to keep name similar to the …

### PR DESCRIPTION
**Pull request by rzsouza**
…original uploaded file

Similar to change uploaded in 4.3.6 only with even more tidying up.

### CODE REVIEW

File: plugins/entando-plugin-jacms/src/main/java/com/agiletec/plugins/jacms/aps/system/services/resource/model/AbstractMultiInstanceResource.java [hunk 1]
```diff
@@ -84,13 +71,12 @@ public boolean isMultiInstance() {
...
-        List<ResourceInstance> instances = new ArrayList<ResourceInstance>(this.getInstances().values());
-        for (int i = 0; i < instances.size(); i++) {
-            ResourceInstance currentInstance = instances.get(i);
+        ResourceDOM resourceDom = getResourceDOM();
+        List<ResourceInstance> resources = new ArrayList<>(instances.values());
+        for (ResourceInstance currentInstance : resources) {
             resourceDom.addInstance(currentInstance.getJDOMElement());
         }
-        Map metadata = getMetadata();
+        Map<String, String> metadata = getMetadata();
```
Comment (joewhite101) [inline plugins/entando-plugin-jacms/src/main/java/com/agiletec/plugins/jacms/aps/system/services/resource/model/AbstractMultiInstanceResource.java]: I wonder about this change. May not matter now but we may need/want metadata to be a tree structure where the values are realized object instances representing xml (or eventually json in the future). Not sure if we have any examples of that now though. 

@eugeniosant Can you confirm that this signature change is safe given what you know about how this gets extended in implementations?

Comment (joewhite101) [reply plugins/entando-plugin-jacms/src/main/java/com/agiletec/plugins/jacms/aps/system/services/resource/model/AbstractMultiInstanceResource.java]: never mind. I see that this matches the signature of the instance in ResourceDom. Looks good.

**Comment by joewhite101** [comment]
Retest this please

**Comment by entando-service** [comment]
All is Well

**Pull request merged**